node/actors/network/src/gossip/mod.rs

```diff
@@ -169,15 +169,27 @@ impl Network {
                 }
             }
 
-            let weight = self.genesis().attesters.weight(
-                &self
-                    .batch_qc
-                    .get(&new_qc.message.number)
-                    .context("last qc")?
-                    .signers,
-            );
+            let weight = self
+                .genesis()
+                .attesters
+                .as_ref()
```

Comment:
extract the attesters once at the beginning of the function and return immediately if not available